### DISCUSSION TITLE
Artifacts for this job instance are unavailable as they may have been purged by Go or deleted externally. Re-run the stage or job to generate them again

### DISCUSSION THREAD

**Discussion by AWS-Viet** [gocd/gocd#9722] [General]
My pipeline ran and done.
But can't see artifacts.
"Artifacts for this job instance are unavailable as they may have been purged by Go or deleted externally. Re-run the stage or job to generate them again."

While I access to go-agent and see all artifacts in pipeline folder of /var/lib

Anyone can help me, pls.
Thanks

**Comment by chadlwilson** [comment]
This is referring to artifact availability on the server rather than the agent. The agent uploads them to the server when your job is complete.

You might want to check your server configuration and whether it is low on space. If it is low and space and configured to auto-delete artifacts, it may be deleting artifacts for any pipelines that don't have "Never Cleanup Artifacts" set, see https://docs.gocd.org/current/configuration/delete_artifacts.html

**Reply by AWS-Viet** [reply]
Of course.
I checked and my pipeline config "**Never Cleanup Artifacts**".
I still check it.

**Reply by AWS-Viet** [reply]
I also check disk of GoCD Server, It is not full.

**Comment by arvindsv** [comment]
Were they ever present? Maybe right after the job is finished.

If nothing, `console.log` should be there. Are you sure nothing external is deleting it? If "Never cleanup artifacts" is enabled, then GoCD should not be deleting anything.

Is this happening for all pipelines or just some?
